src/test/java/teammates/storage/sqlapi/FeedbackResponseCommentsDbTest.java

```diff
@@ -0,0 +1,135 @@
+
+        mockHibernateUtil.verify(() -> HibernateUtil.persist(comment));
+    }
+
+    @Test
+    public void testCreateComment_commentAlreadyExists_throwsEntityAlreadyExistsException() {
+        FeedbackResponseComment comment = getTypicalResponseComment(TYPICAL_ID);
+
+        mockHibernateUtil.when(() -> HibernateUtil.get(FeedbackResponseComment.class, TYPICAL_ID)).thenReturn(comment);
+
+        EntityAlreadyExistsException ex = assertThrows(EntityAlreadyExistsException.class,
+                () -> feedbackResponseCommentsDb.createFeedbackResponseComment(comment));
+
+        assertEquals("Trying to create an entity that exists: " + comment.toString(), ex.getMessage());
+        mockHibernateUtil.verify(() -> HibernateUtil.persist(comment), never());
+    }
+
+    @Test
+    public void testGetComment_commentAlreadyExists_success() {
+        FeedbackResponseComment comment = getTypicalResponseComment(TYPICAL_ID);
+
+        mockHibernateUtil.when(() -> HibernateUtil.get(FeedbackResponseComment.class, TYPICAL_ID)).thenReturn(comment);
+
+        FeedbackResponseComment commentFetched = feedbackResponseCommentsDb.getFeedbackResponseComment(TYPICAL_ID);
+
+        mockHibernateUtil.when(() -> HibernateUtil.get(FeedbackResponseComment.class, TYPICAL_ID)).thenReturn(comment);
+        assertEquals(comment, commentFetched);
+    }
+
+    @Test
+    public void testGetComment_commentDoesNotExist_returnsNull() {
+        mockHibernateUtil.when(() -> HibernateUtil.get(FeedbackResponseComment.class, NOT_TYPICAL_ID)).thenReturn(null);
+
+        FeedbackResponseComment commentFetched = feedbackResponseCommentsDb.getFeedbackResponseComment(NOT_TYPICAL_ID);
+
+        mockHibernateUtil.verify(() -> HibernateUtil.get(FeedbackResponseComment.class, NOT_TYPICAL_ID), times(1));
+        assertNull(commentFetched);
+    }
+
+    @Test
+    public void testDeleteComment_commentExists_success() {
+        FeedbackResponseComment comment = getTypicalResponseComment(TYPICAL_ID);
+
+        mockHibernateUtil.when(() -> HibernateUtil.get(FeedbackResponseComment.class, TYPICAL_ID)).thenReturn(comment);
+        feedbackResponseCommentsDb.deleteFeedbackResponseComment(TYPICAL_ID);
+
+        mockHibernateUtil.verify(() -> HibernateUtil.remove(comment));
+    }
+
+    @Test
+    public void testUpdateComment_commentInvalid_throwsInvalidParametersException() {
+        FeedbackResponseComment comment = getTypicalResponseComment(TYPICAL_ID);
+        comment.setGiverType(FeedbackParticipantType.SELF);
+
+        assertThrows(InvalidParametersException.class,
+                () -> feedbackResponseCommentsDb.updateFeedbackResponseComment(comment));
+
+        mockHibernateUtil.verify(() -> HibernateUtil.merge(comment), never());
+    }
+
+    @Test
+    public void testUpdateComment_commentDoesNotExist_throwsEntityDoesNotExistException() {
+        FeedbackResponseComment comment = getTypicalResponseComment(TYPICAL_ID);
+        comment.setId(101L);
```

Comment:
Should this be `notTypicalId`?

Reply:
Good catch as always, apologies for missing it out!